index.ts

```diff
@@ -3,12 +3,25 @@ import * as sleep from "sleep";
 import * as request from "request-promise-native";
 import * as config from "config";
 import { BigNumber } from "bignumber.js";
+import { U256 } from "codechain-sdk/lib/core/U256";
 
 interface Account {
     address: string;
     balance: BigNumber;
 }
 
+async function calculateNonce(sdk: SDK, payer: string): Promise<U256> {
+    const prevNonce = await sdk.rpc.chain.getNonce(payer);
+    const pendingParcels = await sdk.rpc.chain.getPendingParcels();
+    const payerParcels = pendingParcels.filter((parcel) => parcel.getSignerAddress().value === payer );
+    const nextNonce = await sdk.rpc.chain.getNonce(payer);
+
+    if (nextNonce > prevNonce) {
```

Comment:
What is the purpose of this code?
`prevNonce` and `nextNonce` are acquired by the same method.
When the condition `nextNonce > prevNonce` became true?

Reply:
When the block is mined before `sdk.rpc.chain.getPendingParcels()` is called, `nextNonce` became bigger than `prevNonce`

Reply:
I changed it more simple